src/Container/ConfigFactory.php

```diff
@@ -20,21 +24,32 @@
 class ConfigFactory
 {
     /**
+     * @psalm-suppress MixedArrayAccess
      * @return Config
      */
     public function create(ContainerInterface $container): ConfigInterface
     {
+        /** @var mixed $config */
```

Comment:
`array<string,mixed>|ArrayAccess<string,mixed>` would be a better annotation imho

Reply:
> As mentioned in another comment, just because you request `InjectorInterface::class` from the container, that **must** not be an instance of the requested class-string.

This can be argued here as well. This code is not even requesting a concrete class but a `string` name without any type relation from the container which could be anything. Especially since laminas-di is a component that does not require a laminas/mezzio project.